client/client.go

```diff
@@ -31,6 +31,9 @@ type Client interface {
 	// Services returns access to the set of services.
 	Services() (services.Services, error)
 
+	// ServiceDiscovery returns service discovery with custom namespaces.
+	ServiceDiscovery(opts services.Options) (services.Services, error)
```

Comment:
Any reason why we have a `ServiceDiscovery` API in addition to the `Services` API? Would it be possible to merge the two into `Services(opts services.Options)` and if the options are the default options and there's no namespace override, we return the default services?

Reply:
Yeah I'd prefer to merge them to keep the interface a bit cleaner, this isn't too difficult I imagine?

Reply:
yeah, shouldn't be too difficult, gonna merge